### PR TITLE
Sort reviews

### PR DESCRIPTION
**Pull request by matt-wigg**
Implements the SortReview component with relevant state changes.
    
    1) Sort by relevant defaults to sorting by helpfulness
    2) Sort by Images now properly adds images to the top of the stack and null values to the bottom
    3) Sort by newest time now correctly filtered to latest first by using new Date()
    
    ** fixed the way state is sorted to not mutate state outside of setState()**

**Trello Ticket** : [LINK](https://trello.com/c/OVNrUq5X/34-create-a-new-filtering-component-that-filers-via-most-relevant-includes-customer-photos-most-recent-most-helpful)

### CODE REVIEW

File: client/components/ReviewsOverview.jsx [hunk 1]
```diff
@@ -6,8 +6,8 @@ const ReviewsOverview = {
   reviewScores: [5, 4, 3, 2, 1],
 
   totalReviewsAverageRating(reviews) {
```
Comment (miraclestyle) [inline client/components/ReviewsOverview.jsx:8]: @Matt-Wigg , you can make your server API more useful, and reduce frontend complexity by moving this functionality to the backend code (either as a computed field of the database query, or computed value of the server functionality). This way, you can reuse the computed value anywhere in your frontend code. 

**You should definitely challenge this argument by trying to answer the question: "Where is the best place to host this logic, and what are the possible pros and cons in that case?"**

File: client/index.jsx [hunk 2]
```diff
@@ -34,41 +33,53 @@ class App extends React.Component {
...
+      }
     }
   }
 
-  // averageRaiting() {
-  //   let totalScore = (reviews.reduce((a, b) => a + (b.rating || 0), 0) / reviews.length);
-  //   totalScore = totalScore.toFixed(1);
-  //   console.log(totalScore);
-  // }
+  sortReviewsBy(value) {
```
Comment (miraclestyle) [inline client/index.jsx:64]: @Matt-Wigg , read my previous comment about computing the review score average.

**Review by miraclestyle — APPROVED** [review]
@Matt-Wigg , keep the great work.

File: client/components/ReviewsOverview.jsx (hunk 1 above)
Comment (matt-wigg) [reply client/components/ReviewsOverview.jsx:8]: I love that you have mentioned this!!!! 

Here is my journal entry for the day: 
> At the moment, I am doing one GET request for all reviews and filtering them front-end. Is this best practice???? Or should I trigger a GET request anytime a user interacts with a filtering/show more event? What are the pros and cons of minimizing database queries?

**Pull request merged**